### PR TITLE
feat(datepicker): use readonly arrays for inputs (#2773)

### PR DESCRIPTION
**Pull request by reduckted**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

Part of #2773.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Tha'ts ok. I've updated the origin issue with description on the TS 3.4 improvements in regards of readonly types:
https://github.com/ng-bootstrap/ng-bootstrap/issues/2773#issuecomment-551453072

File: src/datepicker/datepicker-navigation-select.ts [hunk 1]
```diff
@@ -41,8 +41,8 @@ import {NgbDatepickerI18n} from './datepicker-i18n';
 export class NgbDatepickerNavigationSelect implements AfterViewChecked {
   @Input() date: NgbDate;
   @Input() disabled: boolean;
-  @Input() months: number[];
-  @Input() years: number[];
+  @Input() months: ReadonlyArray<number>;
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation-select.ts]: Let's drop this as a non-public API

File: src/datepicker/datepicker-navigation.ts [hunk 2]
```diff
@@ -45,11 +45,11 @@ export class NgbDatepickerNavigation {
 
   @Input() date: NgbDate;
   @Input() disabled: boolean;
-  @Input() months: MonthViewModel[] = [];
+  @Input() months: ReadonlyArray<MonthViewModel> = [];
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation.ts]: Let's drop this as a non-public API

File: src/datepicker/datepicker-navigation.ts [hunk 3]
```diff
@@ -45,11 +45,11 @@ export class NgbDatepickerNavigation {
...
   @Input() showSelect: boolean;
   @Input() prevDisabled: boolean;
   @Input() nextDisabled: boolean;
-  @Input() selectBoxes: {years: number[], months: number[]};
+  @Input() selectBoxes: {years: ReadonlyArray<number>, months: ReadonlyArray<number>};
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation.ts]: Let's drop this as a non-public API

File: src/datepicker/hebrew/ngb-calendar-hebrew.ts [hunk 4]
```diff
@@ -64,7 +64,7 @@ export class NgbCalendarHebrew extends NgbCalendar {
     return day === 0 ? 7 : day;
   }
 
-  getWeekNumber(week: NgbDate[], firstDayOfWeek: number) {
+  getWeekNumber(week: ReadonlyArray<NgbDate>, firstDayOfWeek: number) {
```
Comment (maxokorokov) [inline src/datepicker/hebrew/ngb-calendar-hebrew.ts]: Could we use the `readonly NgbDate[]` syntax for all related `ngb-calendar-XXX.ts` files ?

**Pull request merged**